plugins/sep/sli/sli-helpers.cc

```diff
@@ -129,6 +129,40 @@ spannedRectangle(boost::dynamic_bitset<> bitmap,
   return rect;
 }
 
+int findBestSegFit(unsigned int nSegments, unsigned int height) {
+  int i = 1;
+  while (i * nSegments <= height) {
+    if (i * 2 * nSegments <= height) {
+      i *= 2;
+      continue;
+    }
+
+    break;
+  }
+
+  return i;
+}
```

Comment:
Isn't this simply the largest power of 2 that is smaller than nSegments/height?

That is, I added at the end of the function, before i is returned:

  
  // alternative computation:
  int k = std::log2(height/nSegments);
  
  assert(static_cast<int>(std::pow(2, k)) == i);

And this hasn't tirggered the assertion.

Reply:
good catch!